classes/class-p4-campaigns.php

```diff
@@ -285,5 +293,138 @@ public function enqueue_admin_assets() {
+				'menu_position'      => null,
+				'menu_icon'          => 'dashicons-megaphone',
+				'supports'           => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt', 'comments' )
+			);
+
+			register_post_type( $this->post_type, $args );
+		}
+
+		/**
+		 * Add metabox for campaign page template selection on campaigns cpt
+		 */
+		public function campaign_page_templates_meta_box() {
+			add_meta_box(
+				'campaigns-page-templates',
+				__( 'Page Templates', 'sitepoint' ),
+				array( $this, 'campaign_page_templates_meta_box_callback' ),
+				'campaigns',
+				'side'
+			);
+		}
+
+		/**
+		 * callback function for campaign page template selection
+		 *
+		 * @param $post
+		 */
+		public function campaign_page_templates_meta_box_callback( $post ) {
+
+			// Add a nonce field so we can check for it later.
+			wp_nonce_field( 'campaign_page_template_nonce', 'campaign_page_template_nonce' );
+
+			$value = get_post_meta( $post->ID, '_campaign_page_template', true );
+
+			$campaign_templates = array(
+				'antarctic' => __( 'Antarctic', 'planet4-master-theme-backend' ),
+				'arctic'    => __( 'Arctic', 'planet4-master-theme-backend' ),
+				'forest'    => __( 'Forest', 'planet4-master-theme-backend' ),
+				'oceans'    => __( 'Oceans', 'planet4-master-theme-backend' ),
+				'oil'       => __( 'Oil', 'planet4-master-theme-backend' ),
+				'plastic'   => __( 'Plastics', 'planet4-master-theme-backend' ),
+			);
+
+	?>
+			<select id="campaign_page_template" name="campaign_page_template">
+				<option value=""><?php _e( 'Select Campaign Template', 'planet4-master-theme-backend' ); ?></option>
+				<?php foreach ( $campaign_templates as $campaign => $campaign_template ) {
+					$selected = $value == $campaign ? 'selected="selected"' : '';
+			?>
+					<option
+					value="<?php echo $campaign; ?>" <?php echo $selected; ?>><?php echo $campaign_template; ?></option>
+					<?php
+		}
+		?>
+			</select>
+			<?php
+		}
+
+		/**
+		 * Save campaigns page template data
+		 *
+		 * @param $post_id
+		 */
+		public function save_campaign_page_templates_meta_box_data( $post_id ) {
+
+			// Check if our nonce is set.
+			if ( ! isset( $_POST['campaign_page_template_nonce'] ) ) {
+				return;
+			}
+
+			// Verify that the nonce is valid.
+			if ( ! wp_verify_nonce( $_POST['campaign_page_template_nonce'], 'campaign_page_template_nonce' ) ) {
+				return;
+			}
+
+			// If this is an autosave, our form has not been submitted, so we don't want to do anything.
+			if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+				return;
+			}
+
+			// Check the user's permissions.
+			if ( ! current_user_can( 'edit_post', $post_id ) ) {
+				return;
+			}
+
+			/* OK, it's safe for us to save the data now. */
+
+			// Make sure that it is set.
+			if ( ! isset( $_POST['campaign_page_template'] ) ) {
+				return;
+			}
+
+			// Sanitize user input.
+			$campaign_page_template = sanitize_text_field( $_POST['campaign_page_template'] );
```

Comment:
`sanitize_text_field()` won't validate that the value is one of the acceptable (hard-coded) values. It may be better to use a common function (mentioned above) and validate with `in_array()`.